include/libtorrent/hasher512.hpp

```diff
@@ -48,7 +48,14 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #elif TORRENT_USE_CRYPTOAPI
 #include <windows.h>
-#include <wincrypt.h>
+#include <wincrypt.h> // CALG_SHA_512 conditionally defined here
+
+#ifdef CALG_SHA_512
+#define TORRENT_USE_CRYPTOAPI_SHA_512 1
```

Comment:
it feels like this really belongs in ``config.hpp``, where all the other configuration options are defined. I can't think of a strong argument though, other than symmetry and ease to find where they're coming from. I suppose one reason is that if any other piece of code in libtorrent would check this macro, it may not be obvious that this header needs to be included